extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaProcessor.java

```diff
@@ -336,25 +330,30 @@ private void handleOpenTracing(BuildProducer<ReflectiveClassBuildItem> reflectiv
                 "io.opentracing.contrib.kafka.TracingConsumerInterceptor"));
     }
 
-    private void handleStrimziOAuth(BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
+    private void handleStrimziOAuth(CurateOutcomeBuildItem curateOutcomeBuildItem,
+            BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
         if (!QuarkusClassLoader.isClassPresentAtRuntime("io.strimzi.kafka.oauth.client.JaasClientOauthLoginCallbackHandler")) {
             return;
         }
 
         reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, true,
                 "io.strimzi.kafka.oauth.client.JaasClientOauthLoginCallbackHandler"));
 
-        reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, true,
-                "org.keycloak.jose.jws.JWSHeader",
-                "org.keycloak.representations.AccessToken",
-                "org.keycloak.representations.AccessToken$Access",
-                "org.keycloak.representations.AccessTokenResponse",
-                "org.keycloak.representations.IDToken",
-                "org.keycloak.representations.JsonWebToken",
-                "org.keycloak.jose.jwk.JSONWebKeySet",
-                "org.keycloak.jose.jwk.JWK",
-                "org.keycloak.json.StringOrArrayDeserializer",
-                "org.keycloak.json.StringListMapDeserializer"));
+        if (curateOutcomeBuildItem.getApplicationModel().getDependencies().stream().anyMatch(
+                x -> x.getGroupId().equalsIgnoreCase("org.keycloak")
+                        && x.getArtifactId().equalsIgnoreCase("keycloak-core"))) {
```

Comment:
I've noticed this pattern a few times in this PR. Isn't there a better/more efficient way to look for an artifact in the dependency set? Since it *is* a set, it should be feasible to do a `contains()` AFAICT, which should run in constant time even for large projects, and the case-insensitivity seems unnecessary as well.

Reply:
Thanks for the suggestion David I will try to improve it.

Reply:
I'm afraid it is not that simple since you only compare one part of the coordinates of a dependency (group id and artifact id) the rest (scope, version, type...) is ignored so the method `contains` will always return `false`. One thing that could be done is to propose a method that groups the dependencies by the pair (groupId, artifactid) but to be efficient, you will need a cache and you will need to invalidate it any time the list of dependencies is updated, it adds some complexity with all the consequences behind. I'm wondering if it is really needed within the context of this PR knowing that it is a global approach that we see all over the place in the Quarkus code base, maybe a dedicated task for it would be more efficient, WDYT?

Reply:
Makes sense; maybe we should think of developing a data structure which is backed by a map that is keyed by group & artifact IDs if this is a common pattern.